acceptance: some changes to the schema change allocator test

vivekmenezes opened the pull request.

vivekmenezes committed:
acceptance: move schema validation code out of runSchemaChanges

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/19189)
<!-- Reviewable:end -->

vivekmenezes committed:
acceptance: query over created_at can be served from the index

this change ensures that the bisection code over the timestamps
can bisect even when some index entries do not point to rows.

also fixed a minor issue with using the correct time boundary on
a timespan.
